File: 프로그래머스/1/42840. 모의고사/모의고사.py
def solution(answers):
    one = [1,2,3,4,5] * 2000
    two = [2,1,2,3,2,4,2,5] * 1250
    three = [3,3,1,1,2,2,4,4,5,5] * 1000
    
    cnt_one = 0
    cnt_two = 0
    cnt_three = 0
    
    ans = []
    
    for i in range(len(answers)):
        if one[i] == answers[i]:
            cnt_one +=1
        if two[i] == answers[i]:
            cnt_two +=1
        if three[i] == answers[i]:
            cnt_three +=1
        
        max_v = max(cnt_one,cnt_two,cnt_three)
    
    if cnt_one >= max_v:
        ans.append(1)
    if cnt_two >= max_v:
        ans.append(2)
    if cnt_three >= max_v:
        ans.append(3)
    
    return ans

File: 프로그래머스/1/42840. 모의고사/모의고사.swift
import Foundation

func solution(_ answers:[Int]) -> [Int] {
    let one = [1,2,3,4,5]
    let two = [2,1,2,3,2,4,2,5]
    let three = [3,3,1,1,2,2,4,4,5,5]
    
    var counts = [1 : 0, 2 : 0, 3 : 0]
    
    for i in 0 ..< answers.count {
        
        if answers[i] == one[i % one.count] {
            counts[1]! += 1
        }
        
        if answers[i] == two[i % two.count] {
            counts[2]! += 1
        }
        
        if answers[i] == three[i % three.count] {
            counts[3]! += 1
        }
    }
    return counts.sorted{ $0.key < $1.key }.filter { $0.value == counts.values.max() }.map { $0.key }
}